Sources/Wrappers/AngouriMath.FSharp/Functions.fs

```diff
@@ -29,59 +29,59 @@ let latex x =
     (parsed x).Latexise()
 
 /// Substitutes the given variable with the given value in the given expression
-let substituted x value expr = (parsed expr).Substitute(parsed x, parsed value)
+let substituted x value expr = (parsed expr).Substitute(parsed x, parsed value).InnerSimplified
 
 /// Returns a multiline string representation of matrix
 let printedMatrix (x: AngouriMath.Entity.Matrix) = x.ToString(true)
 
 /// Returns a matrix modified according to the modifier
 let modifiedMatrix (x: AngouriMath.Entity.Matrix) m =
-    x.With(new System.Func<int, int, AngouriMath.Entity, AngouriMath.Entity>(m))
+    x.With(new System.Func<int, int, AngouriMath.Entity, AngouriMath.Entity>(m)).InnerSimplified
```

Comment:
You need to return Matrix here. Use `.Downcast<Entity.Matrix>()` extension from Honk# or downcast via F#'s syntax. F# syntax would look like
```fs
let myFunction ... : Entity.Matrix = downcast ...
```

Reply:
Stop being weird. Use `:?> Entity.Matrix`.